apps/src/code-studio/components/progress/LessonProgress.jsx

```diff
@@ -156,6 +177,11 @@ class LessonProgress extends Component {
               if (isCurrent && level.kind === LevelKind.assessment) {
                 isCurrent = currentPageNumber === level.pageNumber;
               }
+              const onBubbleClick =
+                currentLevelApp === 'music' && level.app === 'music'
```

Comment:
Or, to potentially make this even more dynamic: we could potentially add a new piece of state on progress that indicates whether progression without reloads is enabled, and we just read that state here. That way, Music Lab can internally fire off an action to enable progression without reloads, and this component doesn't need to ever know about the level/game type

Reply:
Agreed that it would be good to generalise.  The nuance here is that the app pairing is important.  For example, if we moved from a "music" to a "maze" level, or a "maze" to a "music" level, we'd want to reload.  We want to ensure that it's only when moving from a "music" to a "music" level in the same lesson that we skip the reload.

Reply:
Ah true, makes sense. Maybe still worth having a flag then so we can check if `currentLevel.app === nextLevel.app` and `currentLevel.progressionWithoutReloadEnabled && nextLevel.progressionWithoutReloadEnabled` to potentially make this available for other labs down the road?

Reply:
I think having per-level flags might be difficult for some potential scenarios down the road.  For example, we might have known pairings of levels that can move between each other.  For that reason, maybe a little helper function that encapsulates the known logic of which pairings work with each other would be appropriately flexible.  (Also, it's proving to be a bunch of work to plumb additional fields all the way down from the server throughout the client, so I'm hesitant to add more fields until we know they're needed.)

Reply:
I have hoisted the check into a small helper function for starters, and we can find a good location for it soon, too.

Reply:
Moved this `canChangeLevelInPage` helper function into `browserNavigation.js`.